Content.Server/LegallyDistinctSpaceFerret/LegallyDistinctSpaceFerretSystem.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System.Linq;
+using System.Numerics;
+using System.Threading;
+using Content.Server.Actions;
+using Content.Server.Atmos.Piping.Unary.Components;
+using Content.Server.Chat.Managers;
+using Content.Server.GameTicking;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.GenericAntag;
+using Content.Server.Ghost.Roles.Components;
+using Content.Server.Interaction;
+using Content.Server.Objectives;
+using Content.Server.Objectives.Systems;
+using Content.Server.Popups;
+using Content.Server.Roles;
+using Content.Server.StationEvents.Events;
+using Content.Shared.Interaction.Components;
+using Content.Shared.Inventory;
+using Content.Shared.LegallyDistinctSpaceFerret;
+using Content.Shared.Mind;
+using Content.Shared.Mobs.Components;
+using Content.Shared.NPC;
+using Content.Shared.Nutrition.EntitySystems;
+using Content.Shared.Objectives.Components;
+using Content.Shared.Popups;
+using Content.Shared.Roles;
+using Robust.Server.Audio;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Timer = Robust.Shared.Timing.Timer;
+
+namespace Content.Server.LegallyDistinctSpaceFerret;
+
+public sealed class LegallyDistinctSpaceFerretSystem : EntitySystem
```

Comment:
Not relevant for an april fools PR (i.e. this is just learning advice!) but as I'm sure you know much of this could be improved, here's some broad advice I'd keep in mind for future note:

- Move objectives to their own systems, one per objective.
- Create a new component and system for managing the mind roles (assuming we have no such thing already)
- Use a collision circle instead of the timer jank, look at how bullets handle their whizzing noise that you hear when they pass by you.
- Move all hardcoded locale strings to fields on the component.
- Move eeping into its own system, have the hibernate condition listen for a broadcasted event that tells it you eeped, and have the role handle the win sfx/etc somehow.
- not throw it all into one big file aaa (understandable in an april fools PR but the usual structure is a parent folder with Components, Systems, Prototypes, etc as subfolders.)
- move backflip to its own system as well.

Reply:
Keeping stuff compressed in big files (e.g. the one big yml/ftl files) is just for April Fools merge safety given I have no idea what chaos the maintainers etc will brew up. Will enact some of the code changes, thanks